frontend/src/views/FinalSupplyEquipments/_schema.jsx

```diff
@@ -0,0 +1,359 @@
+import { Typography } from '@mui/material'
+import moment from 'moment'
+import { v4 as uuid } from 'uuid'
+
+// Copy the desired columns to new row
+const duplicateRow = (props) => {
+  const newRow = {
+    ...props.data,
+    id: uuid(),
+    finalSupplyEquipmentId: null,
+    serialNbr: undefined,
+    latitude: undefined,
+    longitude: undefined,
+    modified: true
+  }
+
+  if (props.api) {
+    props.api.applyTransaction({
+      add: [newRow],
+      addIndex: props.node?.rowIndex + 1
+    })
+    props.api.stopEditing()
+  } else {
+    console.error('API is undefined')
+  }
+}
+
+const deleteRow = (props) => {
+  const updatedRow = { ...props.data, deleted: true, modified: true }
+  if (props.api) {
+    props.api.applyTransaction({ update: [updatedRow] })
+    props.api.applyTransaction({ remove: [props.node.data] })
+    props.api.stopEditing()
+  } else {
+    console.error('API is undefined')
+  }
+}
+
+export const finalSupplyEquipmentColDefs = (
+  t,
+  optionsData,
+  compliancePeriod,
+  api
+) => [
+  {
+    colId: 'action',
+    cellRenderer: 'actionsRenderer',
+    cellRendererParams: {
+      enableDuplicate: true,
+      enableEdit: false,
+      enableDelete: true,
+      onDuplicate: (props) => duplicateRow({ ...props, api }),
+      onDelete: (props) => deleteRow({ ...props, api })
+    },
+    pinned: 'left',
+    maxWidth: 150,
+    editable: false,
+    suppressKeyboardEvent,
+    filter: false,
+    cellStyle: { backgroundColor: '#f2f2f2' }
+  },
+  {
+    field: 'id',
+    cellEditor: 'agTextCellEditor',
+    cellDataType: 'text',
+    hide: true
+  },
+  {
+    field: 'complianceReportId',
+    headerName: t(
+      'finalSupplyEquipment:finalSupplyEquipmentColLabels.complianceReportId'
+    ),
+    cellEditor: 'agTextCellEditor',
+    cellDataType: 'text',
+    hide: true
+  },
+  {
+    field: 'supplyFrom',
+    headerName: t(
+      'finalSupplyEquipment:finalSupplyEquipmentColLabels.supplyFrom'
+    ),
+    headerComponent: 'headerComponent',
+    width: 330,
+    cellRenderer: (params) => (
+      <Typography variant="body4">
+        {params.value[0]
+          ? `${params.value[0]} to ${params.value[1]}`
+          : 'YYYY-MM-DD to YYYY-MM-DD'}
+      </Typography>
+    ),
+    suppressKeyboardEvent,
+    cellStyle: (params) => {
+      if (params.data.modified && (!params.value || params.value === ''))
+        return { borderColor: 'red' }
+    },
+    cellEditor: 'dateRangeCellEditor',
+    cellEditorParams: {
+      minDate: moment(`${compliancePeriod}-01-01`, 'YYYY-MM-DD').toDate(),
+      maxDate: moment(`${compliancePeriod}-12-31`, 'YYYY-MM-DD').toDate()
+    },
+    valueGetter: (params) => {
+      return [params.data.supplyFromDate, params.data.supplyToDate]
+    },
+    valueSetter: (params) => {
+      params.data.supplyFromDate = params.newValue[0]
+      params.data.supplyToDate = params.newValue[1]
+      return true
+    }
+  },
+  // {
```

Comment:
commented code needed?

Reply:
Yes, I used the date range selector instead of two separate dates as given in the UX design. So after confirming from Al about which one goes well here, will delete the redundant code.